multi/imxrt-multi/uart.c

```diff
@@ -142,21 +220,37 @@ static void uart_intrThread(void *arg)
 	uart_t *uart = (uart_t *)arg;
 	uint8_t mask;
 	uint8_t c;
+	int rxActive = 1;
 
 	for (;;) {
 		/* wait for character or transmit data */
 		mutexLock(uart->lock);
 		while (lf_fifo_empty(&uart->rxFifoCtx) != 0) { /* nothing to RX */
-			if (libtty_txready(&uart->tty_common)) { /* something to TX */
+			uint32_t intrs = (1 << 27) | (1 << 26) | (1 << 25) | (1 << 21);
+			if (libtty_txready(&uart->tty_common)) {          /* something to TX */
 				if (uart_getTXcount(uart) < uart->txFifoSz) { /* TX ready */
 					break;
 				}
 				else {
-					*(uart->base + ctrlr) |= 1 << 23;
+					intrs |= 1 << 23;
 				}
 			}
 
-			*(uart->base + ctrlr) |= (1 << 27) | (1 << 26) | (1 << 25) | (1 << 21);
+			if ((uart->halfDuplexAction.port != 0) && (rxActive == 0)) {
+				uint32_t status = *(uart->base + statr);
+				if ((status & (1 << 22)) != 0u) {
+					/* Transmission finished, activate RX */
+					gpio_setPin(&uart->halfDuplexAction, 0);
+					*(uart->base + ctrlr) |= (1 << 18);
```

Comment:
Is there some reason for not setting it through `intrs`?

Reply:
Because this bit is not interrupt related - I wanted `intrs` to specifically contain bits that turn on interrupts.